controllers/utils/manifestworkreplicaset.go

```diff
@@ -0,0 +1,240 @@
+	return generateManifestWorkReplicaset(name, namespace, expectedValueAnn, jsonPaths, ibu)
+}
+
+// GenerateRollbackManifestWorkReplicaset returns a populated ManifestWorkReplicaSet for rollback stage of an ImageBasedUpgrade
+func GenerateRollbackManifestWorkReplicaset(name, namespace string, ibu *lcav1alpha1.ImageBasedUpgrade) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
+	ibu.Spec.Stage = lcav1alpha1.Stages.Rollback
+	jsonPaths := []mwv1.JsonPath{
+		{
+			Name: "isRollbackCompleted",
+			Path: `.status.conditions[?(@.type=="RollbackCompleted")].status`,
+		},
+		{
+			Name: "rollbackInProgressConditionMessage",
+			Path: `.status.conditions[?(@.type=="RollbackInProgress")].message'`,
+		},
+		{
+			Name: "rollbackCompletedConditionMessages",
+			Path: `.status.conditions[?(@.type=="RollbackCompleted")].message`,
+		},
+	}
+	expectedValueAnn := fmt.Sprintf(manifestWorkExpectedValuesAnnotationTemplate, "isRollbackCompleted")
+	return generateManifestWorkReplicaset(name, namespace, expectedValueAnn, jsonPaths, ibu)
+}
+
+// GeneratePrepManifestWorkReplicaset returns a populated ManifestWorkReplicaSet for Prep stage of an ImageBasedUpgrade
+func GeneratePrepManifestWorkReplicaset(name, namespace string, ibu *lcav1alpha1.ImageBasedUpgrade) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
+	ibu.Spec.Stage = lcav1alpha1.Stages.Prep
+	jsonPaths := []mwv1.JsonPath{
+		{
+			Name: "isPrepCompleted",
+			Path: `.status.conditions[?(@.type=="PrepCompleted")].status`,
+		},
+		{
+			Name: "prepInProgressConditionMessage",
+			Path: `.status.conditions[?(@.type=="PrepInProgress")].message'`,
+		},
+		{
+			Name: "prepCompletedConditionMessages",
+			Path: `.status.conditions[?(@.type=="PrepCompleted")].message`,
+		},
+	}
+	expectedValueAnn := fmt.Sprintf(manifestWorkExpectedValuesAnnotationTemplate, "isPrepCompleted")
+	return generateManifestWorkReplicaset(name, namespace, expectedValueAnn, jsonPaths, ibu)
+}
+
+func generateManifestWorkReplicaset(name, namespace, expectedValueAnn string, jsonPaths []mwv1.JsonPath, ibu *lcav1alpha1.ImageBasedUpgrade) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
+	ibuRaw, err := ibuToBytes(ibu)
+	if err != nil {
+		return nil, err
+	}
+	manifestConfigs := []mwv1.ManifestConfigOption{
+		{
+			ResourceIdentifier: mwv1.ResourceIdentifier{
+				Name:     ibu.GetName(),
+				Group:    ibu.GroupVersionKind().Group,
+				Resource: ibu.GetResourceVersion(),
+			},
+			FeedbackRules: []mwv1.FeedbackRule{
+				{
+					Type:      mwv1.JSONPathsType,
+					JsonPaths: jsonPaths,
+				},
+			},
+		},
+	}
+	mwrs := &mwv1alpha1.ManifestWorkReplicaSet{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				manifestWorkExpectedValuesAnnotation: expectedValueAnn,
+			},
+		},
+		Spec: mwv1alpha1.ManifestWorkReplicaSetSpec{
+			PlacementRefs: []mwv1alpha1.LocalPlacementReference{
+				{
+					Name: "dummy",
+				},
+			},
```

Comment:
Without knowing anything about the ManifestWorkReplicaSetSpec....curious why we need this one LocalPlacementReference placeholder? 

sidenote: maybe `dummy` --> `placeholder`

Reply:
If we were to actually populate this programatically, `placeholder` would make sense. In our case, we don't actually use the ManifestWorkReplicaSet controller to rollout the manifests. TALM does it based on the things defined in CGU. That's why I called it dummy in my test files. What's missing here is another ManifestWorkTemplate CRD. Do you think it's better if we define it in our API group instead of abusing ManifestWorkReplicaSet? We will have to decide before making TALM manifestwork support GA.

Reply:
Ahh just read up on it a bit. So basically if i understanding this correctly...ManifestWorkReplicaSet basically means placementRefs + manifestWorkTemplate

But since we dont use the placementRefs feature (CGU is handling that instead) that just leaves us with only the manifestWork to modify but we need the PlacementRefs is a required field? 


> What's missing here is another ManifestWorkTemplate CRD. Do you think it's better if we define it in our API group instead of abusing ManifestWorkReplicaSet

I really want to say yes. Would that be too much work? So basically we would owns() ManifestWork instead ManifestWorkReplicaSet?

Reply:
We'll probably call it ManifestWorkTemplate. It would have to be under our lcm api group. But the definition should be mostly imported from OCM, plus the extension we add for the expected values. Not that it's hard. It's just not ideal to have something really similar but not exactly the same as the OCM CRDs. Ideally, we need the OCM manifest work to support expected values natively (side by side with the feedback rules and jsonpath) and expose the template part as a CRD instead of a subsection of the MWRS.